okhttp/src/main/java/okhttp3/internal/http2/Http2Codec.java

```diff
@@ -195,28 +194,18 @@ public Http2Codec(OkHttpClient client, Interceptor.Chain chain, StreamAllocation
   }
 
   class StreamFinishingSource extends ForwardingSource {
-    private boolean completed = false;
+    boolean completed = false;
 
     StreamFinishingSource(Source delegate) {
       super(delegate);
     }
 
-    @Override public long read(Buffer sink, long byteCount) throws IOException {
-      try {
-        return super.read(sink, byteCount);
-      } catch (IOException ioe) {
-        completed = true;
-        streamAllocation.eventListener.responseBodyEnd(streamAllocation.call, ioe);
```

Comment:
Did you mean to get rid of this? Seems like we won't call the `responseBodyEnd` on an IOException for HTTP/2.